backend/local/backend_plan.go

```diff
@@ -162,19 +162,20 @@ func (b *Local) opPlan(
 
 		if plan.Changes.Empty() {
 			b.CLI.Output("\n" + b.Colorize().Color(strings.TrimSpace(planNoChanges)))
-			return
+		} else {
```

Comment:
This new approach means that everything after this point will need to remember to guard for `plan.Changes.Empty()`, which seems likely to lead to mistakes as this code evolves.

What do you think about instead just adding a new call to `b.ShowDiagnostics` just before the `return`, so it'll still happen in both codepaths but the rest of this function can assume there are changes to make?

Reply:
@apparentlymart sure, thx, updated